platform/msm8974/gpio.c

```diff
@@ -52,6 +52,11 @@ void gpio_set(uint32_t gpio, uint32_t dir)
 	return;
 }
 
+uint32_t gpio_status(uint32_t gpio)
```

Comment:
Hm interesting, For the future lk2nd rebase we have implemented a different way to use gpios to avoid things like this.

Any chance you can try to add your device on top of the `experimental-tmp4` branch? The dts model is a bit different, but that branch (and also `rebase/dts`) has some examples.

Reply:
OK I will try to use that branch as a base, will let you know when I get it to work.

Reply:
I tried the `experimental-tmp4` branch, and the new gpio driver works great, no change required.

I did have to rollback to the older (current) version of `mkbootimg`. The new version in that branch did not generate a working image.

In any case, I don't know what is the plan for that new codebase, I can wait for it to be merged and in use, and resend a new PR. I will fix the other comments in that new PR.

Reply:
We want to eventually switch to the new branch as a primary one, carrying over all the devices we support here.

It's not an issue for dts-only changes but I want to make sure all new code changes are addressed on the new branch, so this PR is "non-trivial" in that regard...

I'll still be happy to accept it here so you have a working bootloader in the meantime, though the changes and crashes are "interesting" here... I suppose all the same changes are necessary there and the crash is still present on the new branch?